openff/toolkit/topology/molecule.py

```diff
@@ -36,21 +36,21 @@
 import operator
 import warnings
 from abc import abstractmethod
-from collections import OrderedDict
+from collections import OrderedDict, UserDict, defaultdict
 from copy import deepcopy
 from typing import Optional, Union
+from itertools import chain
 
+
+import json
 import networkx as nx
 import numpy as np
+from cached_property import cached_property
```

Comment:
```suggestion
from functools import cached_property
```
?

Reply:
This may be related to `cached_property` only being available in 3.8

Reply:
Looks like you're usiing https://pypi.org/project/cached-property/ I'm not sure if https://pypi.org/project/backports.cached-property/ would be better, and the answer comes from how each works with 3.8+; a solution in the standard library should be used over a third-party backport. I have not looked into this but it should be a quick check.